contrib: Add zsh completion scripts

mstampfer opened the pull request:
Adds zsh completion support using the compdef system for:
  - bitcoin-cli with RPC command completion and context-sensitive arguments
  - bitcoin-tx with option/command completion and file path handling
  - bitcoind with configuration option completion

  These complement the existing bash completion scripts and provide
  the same functionality for zsh users.

mstampfer commented:
This PR adds zsh completion scripts for Bitcoin Core tools, converting the existing bash completions to work with zsh's compdef system. This Improves user experience for zsh users by addressing the need for tab completion support for zsh which doesn't rely on the existing bash completion scripts.

The implementation maintains the same functionality as the bash versions while using zsh-native completion functions. 

I'm happy to address any concerns or make requested changes.

mstampfer commented:
I have opened an [Issue requesting zsh completion support](https://github.com/bitcoin/bitcoin/issues/33404). Please reopen this PR or please comment with any concerns why this feature should not be merged.

maflcko reviewed (commented):
You'd have to update the line in the docs? https://github.com/bitcoin/bitcoin/blob/3b5e6121f58f0ca4663deabe793a26be38b0f535/contrib/README.md?plain=1#L45

Commit:
contrib: Add zsh completion scripts for bitcoin tools

- Add zsh completion scripts for bitcoin-cli, bitcoin-tx, and bitcoind
- Update README.md to mention zsh completions alongside bash and fish
- Proper copyright headers and formatting for all zsh completion files
